src/dbvt/mod.rs

```diff
@@ -754,17 +746,14 @@ where
             if parent_parent_index == 0 {
                 self.root_index = sibling_index;
             } else {
-
                 // else we have a remaining branch, and need to update either left or right to point
                 // to the sibling, based on where the old branch node was
                 match self.nodes[parent_parent_index] {
-                    Node::Branch(ref mut b) => {
-                        if b.left == parent_index {
-                            b.left = sibling_index;
-                        } else {
-                            b.right = sibling_index;
-                        }
-                    }
+                    Node::Branch(ref mut b) => if b.left == parent_index {
```

Comment:
this also doesn't look better

Reply:
Agree, and there's a config switch for it.